leopard.go

```diff
@@ -426,19 +430,19 @@ func (r *leopardFF16) reconstruct(shards [][]byte, recoverAll bool) error {
 	for i := 0; i < r.parityShards; i++ {
 		if len(shards[i+r.dataShards]) == 0 {
 			errLocs[i] = 1
-			if LEO_ERROR_BITFIELD_OPT && recoverAll {
+			if LEO_ERROR_BITFIELD_OPT && (recoverAll || (required != nil && required[i])) {
 				errorBits.set(i)
 			}
 		}
 	}
 	for i := r.parityShards; i < m; i++ {
 		errLocs[i] = 1
-		if LEO_ERROR_BITFIELD_OPT && recoverAll {
+		if LEO_ERROR_BITFIELD_OPT && (recoverAll || (required != nil && required[i])) {
 			errorBits.set(i)
 		}
 	}
 	for i := 0; i < r.dataShards; i++ {
-		if len(shards[i]) == 0 {
+		if len(shards[i]) == 0 /*&& (recoverAll || (required != nil && required[i]))*/ {
```

Comment:
Uncommenting this leads to more significant improvements in `BenchmarkDecode1K` but will lead to a bunch of test failures (I think mostly because `ReconstructData` expects to reconstruct all original missing data). This can probably be fixed.